tied-buffer/sycl-2.2/buffer-tags.md

```diff
@@ -54,14 +63,15 @@ class buffer {
 With this interface, a user can specify a specialisation of the buffer class by simply providing one or more values of the following tag enum class to the constructor of a buffer.
 
 ```cpp
-enum class tag {
-  context_bound,
-  mapping,
-  svm
+namespace property {
+  struct map_ptr;
+  struct cl_interop;
+  struct gl_interop;
+  struct svm;  // SYCL 2.2 Only
 };
 ```
 
-Some of these tags will require an additional interface for the buffer class, for example, the `tag::context_bound` tag will need a `get_context()` member function.
+Some of these tags will require an additional interface for the buffer class, for example, the `property::context_bound` tag will need a `get_context()` member function.
```

Comment:
yes that's right, though in most cases the property object will be very lightweight so shouldn't incur any significant overhead.

We would also have to require that calling the `get_property` member function with a property type that the buffer does not have, throws an exception. Which should be reasonable as we also have the `has_property` member function to query if the buffer does have the property.

Yes, I can prototype this.